pulser/sequence.py

```diff
@@ -375,14 +528,72 @@ def align(self, *channels):
         if len(channels) < 2:
             raise ValueError("Needs at least two channels for alignment.")
 
+        if not self._building:
+            return
+
         last_ts = {id: self._last(id).tf for id in channels}
         tf = max(last_ts.values())
 
         for id in channels:
             delta = tf - last_ts[id]
             if delta > 0:
                 self.delay(delta, id)
+                self._calls.pop()
 
+    def build(self, **vars):
+        """Builds a sequence from the programmed instructions.
+
+        Keyword Args:
+            vars: The values for all the variables declared in this Sequence
+                instance, indexed by the name given upon declaration. Check
+                `Sequence.declared_variables` to see all the variables.
+
+        Returns:
+            Sequence: The Sequence built with the given variable values.
+
+        Example:
+            # Check which variables are declared
+            >>> print(seq.declared_variables)
+            {'x': Variable(name='x', dtype=<class 'float'>, size=1),
+             'y': Variable(name='y', dtype=<class 'int'>, size=3)}
+            # Build a sequence with specific values for both variables
+            >>> seq1 = seq.build(x=0.5, y=[1, 2, 3])
+        """
+        if self._building:
+            warnings.warn("Building a non-parametrized sequence simply returns"
+                          " a copy of itself.")
+            return copy.copy(self)
+        all_keys, given_keys = self._variables.keys(), vars.keys()
+        if given_keys != all_keys:
+            invalid_vars = given_keys - all_keys
+            if invalid_vars:
+                warnings.warn("No declared variables named: "
+                              + ", ".join(invalid_vars))
+                for k in invalid_vars:
+                    vars.pop(k, None)
+            missing_vars = all_keys - given_keys
+            if missing_vars:
+                raise TypeError("Did not receive values for variables: "
+                                + ", ".join(missing_vars))
+
+        for name, value in vars.items():
+            self._variables[name]._assign(value)
+
+        # Can't deepcopy with stored parametrized objects due to recursiveness
+        seq = copy.copy(self)
```

Comment:
Why do you need to copy here since you are already deepcopying below ? Wouldn't a dummy `seq = Sequence()` be enough here ?

Reply:
Good question. I'm copying so that I start from how the sequence was before the first `Variable` was declared (when `self._building` switches to `False`). Also, channel declarations are not stored, so they always come along here. 

However, there's also the possibility of starting from the empty `Sequence` and executing the calls stored in `_calls`. This would amount to recreating the object from the start, the same way it'll be done when you deserialize it.